### ISSUE TITLE
[Refactor][Attention] decode ops pad K/V to a compile-time capacity before calling the kernel

### ISSUE THREAD

**Issue by lcy-seso** [tile-ai/TileOPs#1863]
## Description

`MultiHeadAttentionDecodeFwdOp.forward` and `GroupedQueryAttentionDecodeFwdOp.forward` pad K/V
along the sequence axis before calling the kernel:

- `tileops/ops/attention/mha.py:204`
- `tileops/ops/attention/gqa.py:1475`

```python
real_seqlen_kv = k.shape[1]
if real_seqlen_kv < self.seqlen_kv:
    k = F.pad(k, pad=(0, 0, 0, 0, 0, self.seqlen_kv - real_seqlen_kv), value=0)
    v = F.pad(v, ...)
return self._get_kernel(q.dtype)(q, k, v, real_seqlen_kv)
```

**Root cause.** The decode kernel declares its K/V buffer with a compile-time extent taken from
the op's constructor, and separately takes the true length as a runtime scalar it masks against:

```python
# tileops/kernels/attention/mha_decode.py:31,41,77
shape_kv = [batch, seqlen_kv, heads, dim]          # compile-time
real_seqlen_kv: T.int32                            # runtime
acc_s[i, j] = T.if_then_else(k * block_N + j < real_seqlen_kv, 0, ...)
```

The kernel already ignores the padded region. The pad exists only to satisfy the buffer's shape
signature, so every short-KV decode call allocates a full-capacity K and V and copies the real
data into them — pure overhead, and the copy grows with the gap between the real length and the
declared capacity.

It is also a layering leak: the op materialises a buffer shaped to one backend's compiled
program. A second backend would inherit "this slot takes pre-padded K/V" as part of its contract.

**The right shape already exists in-tree.** `MultiHeadAttentionDecodePagedWithKVCacheFwdOp`
(`mha.py`) and its GQA counterpart take a physical cache plus `real_seqlen_kv` and a
`block_table`, and never pad. Only the non-paged decode ops leak.

**Reproduce**

```python
import torch
from tileops.ops import MultiHeadAttentionDecodeFwdOp

op = MultiHeadAttentionDecodeFwdOp(batch=1, heads=8, seqlen_q=1, seqlen_kv=4096, dim=64)
q = torch.randn(1, 1, 8, 64, device="cuda", dtype=torch.float16)
k = torch.randn(1, 128, 8, 64, device="cuda", dtype=torch.float16)   # 128 << 4096
v = torch.randn_like(k)
op(q, k, v)   # allocates and fills two [1, 4096, 8, 64] tensors to run a 128-length decode
```

## Goal

Decode ops hand the kernel the K/V they were given. The declared capacity stops being a shape the
caller has to materialise.

## Plan

This needs a route decided before implementation; the three are not variations of one change.

1. **Dynamic KV extent in the kernel.** Make the extent symbolic so the buffer signature accepts
   the real length. Touches `mha_decode.py`, `gqa_decode.py`, `gqa_decode_bs1.py` and their split
   variants; the loop bound `T.ceildiv(seqlen_kv, block_N)` becomes a runtime value. Needs a
   benchmark: a dynamic bound may cost codegen quality on the hot path.
2. **Op-owned persistent buffer.** Allocate the padded K/V once per op and copy into it. Removes
   the repeated allocation but keeps the copy and keeps padding in the op layer, so the layering
   leak stays.
3. **Retire the non-paged decode ops.** The paged variants already express the dynamic contract.
   This is a public-surface decision, not a kernel change.

Route 3 makes 1 and 2 unnecessary. Decide the route first, then open implementation work against
it.

## Constraints

- No change to decode numerics or to which kernel a call selects.
- A benchmark is required for route 1 before it lands: the extent feeds a loop bound.
- The paged decode path must keep working unchanged whichever route is taken.

## Acceptance Criteria

- [ ] A route is chosen and recorded, with the reason the other two were rejected
- [ ] No `F.pad` remains in `tileops/ops/attention/`
- [ ] Decode correctness holds for `real_seqlen_kv` well below the declared capacity, across dtypes
- [ ] Decode throughput does not regress against the current padded path
